### PR TITLE
feat(angular): add angular wrapper

### PR DESCRIPTION
**Pull request by ls-infra**
add the `MonacoAngularWrapperComponent` as the equivalent to `packages/wrapper-react/src/index.tsx`
so that the Angular devs can refer this to do two ways data binding (load from API and save back to API ,which is the typical use of the editor )
aslo by passing the `wrapperConfig` into the wrapper to make it language agnostic to increase reusability 

but need some help for this run time exception atm  
image

thank you very much

### CODE REVIEW

**Comment by ls-infra** [comment]
this wrapper enhances and relates to https://github.com/TypeFox/monaco-languageclient/pull/759 which changes the component to `standalone` to make it more reusable and configurable .

File: verify/angular/src/app/app.component.ts [hunk 1]
```diff
@@ -3,30 +3,32 @@
...
  * ------------------------------------------------------------------------------------------ */
 
-import { AfterViewInit, Component } from '@angular/core';
-import { MonacoEditorLanguageClientWrapper } from 'monaco-editor-wrapper';
+import { AfterViewInit, Component, signal } from '@angular/core';
+import { WrapperConfig } from 'monaco-editor-wrapper';
+import { MonacoAngularWrapperComponent } from '../monaco-angular-wrapper.component';
 import { buildJsonClientUserConfig } from 'monaco-languageclient-examples/json-client';
-
+import '@codingame/monaco-vscode-groovy-default-extension';
```
Comment (kaisalmen) [inline verify/angular/src/app/app.component.ts:10]: This is not needed needed here.

File: verify/angular/src/monaco-angular-wrapper.component.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+.monaco-editor {
+  height: 50vh;
+}
+`,
+})
+export class MonacoAngularWrapperComponent implements AfterViewInit, OnDestroy {
+    @Output() onTextChanged = new EventEmitter<string>();
+    @Input() wrapperConfig: WrapperConfig;
+    title = 'angluar-lang-client';
+    private wrapper: MonacoEditorLanguageClientWrapper =
```
Comment (kaisalmen) [inline verify/angular/src/monaco-angular-wrapper.component.ts:39]: You don't have to wrap at 80 characters.

**Review by kaisalmen — COMMENTED** [review]
Some small remarks. Regarding the error you saw, I have to test this. Is the dom element (the htmlContainer) defined?

**Comment by kaisalmen** [comment]
Yes, the div element with id "monaco-editor-root" is no longer set.

Btw, I switched to the esbuild based system in the angular example, because the workers where no longer working with the custom webpack config. It was able to fix it in the independent webpack example, but that does not work in the angular context.

Due to esbuild/vite build currently some resources from monaco-vscode-api can't be loaded. @CGNonofr FYI, I hoped your existing vite plugin can help here, but the problem origin seems to be different or I need an asset config? I am not super familiar with Angular configuration. If someone can tell me how I can influence/change the vite config in Angular 18 I will be very happy. esbuild plugins can be used. 
image

**Comment by ls-infra** [comment]
image
@kaisalmen ,  i just finished the change , 
still some error , but looks like it is working ,
thanks

**Comment by kaisalmen** [comment]
@ls-infra thank you, I was away for a week. I will also try to find a solution for the above error. I will provide feedback in the upcoming days.

**Comment by ls-infra** [comment]
@kaisalmen 
i think for the purpose of this PR , it is ready to merge now,  the demo shows that it allows user to edit the code and save to  an API.  i just did the `git rebase`

the errors in the console may be an exiting issue that we can track in another PR. 

image

thanks

**Comment by ls-infra** [comment]
@kaisalmen i am thinking to send another PR to fix that console error issue and support multiple lang and theme selection in the wrapper demo like the  https://microsoft.github.io/monaco-editor/.

so that it will be easy to allow the dev user to test different lang easily . (while currently is only for json , when i tried to bind it to groovy, it faces some theme.json issue again for client side syntax highlight )

image

**Review by kaisalmen — APPROVED** [review]
Thank you.

Sorry, you have to rebase once more. I didn't think about the impact of the `-next.5` release.

This is basically ready for merge. My comments are non blocking.

Are you willing to maintain this specific example in the future?

File: packages/examples/src/json/server/main.ts [hunk 3]
```diff
@@ -23,4 +25,22 @@ export const runJsonServer = (baseDir: string, relativeDir: string) => {
             perMessageDeflate: false
         }
     });
+
+    startMockHttpServerForSavingCodeFromEditor();
+};
+
+export const startMockHttpServerForSavingCodeFromEditor = () => {
```
[the remark this answers is not in the document]
Comment (ls-infra) [reply packages/examples/src/json/server/main.ts:32]: this is a full stack demo to demo how to save code asynchronously from editor to API.
Yes I am willing to maintain the examples.

File: verify/angular/src/monaco-angular-wrapper/monaco-angular-wrapper.component.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    async ngOnDestroy() {
+        this.destroyMonaco();
+    }
+
+    handleOnTextChanged() {
+        const textModels = this.wrapper.getTextModels();
+        if (textModels) {
+            const verifyModelContent = () => {
```
[the remark this answers is not in the document]
Comment (ls-infra) [reply verify/angular/src/monaco-angular-wrapper/monaco-angular-wrapper.component.ts]: i updated pushed the change for using `didModelContentChange` and git rebase . 
the CI is passed , so i will merge it now , we will open new PRs for the other related things. 

thx

**Comment by ls-infra** [comment]
@kaisalmen  the CI is passed after the git rebase ,
 can you merge it now , 
we will open new PRs for the other related things. 
thx

File: verify/angular/src/monaco-angular-wrapper/monaco-angular-wrapper.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kaisalmen) [reply verify/angular/src/monaco-angular-wrapper/monaco-angular-wrapper.component.ts]: Thank you

File: packages/examples/src/json/server/main.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/examples/src/json/server/main.ts:32]: That is good news!

**Comment by kaisalmen** [comment]
I squashed merged your commits. Thank you for providing this. Let's continue from here. 🎉

**Pull request merged**